### PR TITLE
Update toolchain and make sure versions are pinned

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: deploy/skaffold/Dockerfile [hunk 1]
```diff
@@ -12,59 +12,91 @@
...
-  --disable-installation-options && \
-  rm -rf google-cloud-sdk-*.tar.gz
+COPY --from=docker:18.09.6 /usr/local/bin/docker /usr/local/bin/
+COPY --from=download-kubectl kubectl /usr/local/bin/
+COPY --from=download-helm helm /usr/local/bin/
+COPY --from=download-kustomize kustomize /usr/local/bin/
+COPY --from=download-kompose kompose /usr/local/bin/
+COPY --from=download-container-structure-test container-structure-test /usr/local/bin/
+COPY --from=download-kind kind /usr/local/bin/
+COPY --from=download-bazel /bazel /usr/local/bin/
```
Comment (balopat) [inline deploy/skaffold/Dockerfile]: just to be consistent can this be `bazel` as well instead of `/bazel` or all the above `/xyz`?

Comment (balopat) [reply deploy/skaffold/Dockerfile]: or I guess for directories this is fine?

Comment (balopat) [reply deploy/skaffold/Dockerfile]: same for google-cloud-sdk

Comment (dgageot) [reply deploy/skaffold/Dockerfile]: done!

**Comment by balopat** [comment]
Thank you, I love the separate targets per download - hope to see this ran on buildkit one day being sped up in parallel!

File: deploy/skaffold/Dockerfile [hunk 2]
```diff
@@ -12,59 +12,91 @@
...
-  --bash-completion=false \
-  --disable-installation-options && \
-  rm -rf google-cloud-sdk-*.tar.gz
+COPY --from=docker:18.09.6 /usr/local/bin/docker /usr/local/bin/
+COPY --from=download-kubectl kubectl /usr/local/bin/
+COPY --from=download-helm helm /usr/local/bin/
+COPY --from=download-kustomize kustomize /usr/local/bin/
+COPY --from=download-kompose kompose /usr/local/bin/
+COPY --from=download-container-structure-test container-structure-test /usr/local/bin/
+COPY --from=download-kind kind /usr/local/bin/
```
Comment (balopat) [inline deploy/skaffold/Dockerfile]: hold on don't we need kind only for integration? we shouldn't have it for runtime_deps

Comment (dgageot) [reply deploy/skaffold/Dockerfile]: I thought it wouldn't hurt :-)

Comment (dgageot) [reply deploy/skaffold/Dockerfile]: fixed!

Comment (balopat) [reply deploy/skaffold/Dockerfile]: I guess it is an interesting option to have in the future. This structure allows for creating some nice "flavors" of skaffold, we could have kind included in them and that could solve provisioning clusters immediately with skaffold...which can be a pretty cool, one button setup: `docker run -v $pwd:/app "--priviliged-and-daemon-mounted" gcr.io/k8s-skaffold/skaffold skaffold up`  that includes skaffold + kind in one go.

**Pull request merged**